### PR TITLE
Added documentation for the output components

### LINKED ISSUE #4: Document Output Elements
You will need to illustrate the various options of individual components and how to use it on CircuitVerse. Tips and tricks will be helpful. 

- [ ] Output
- [ ] RGBLed
- [ ] LED
- [ ] VariableLed
- [ ] HexDisplay
- [ ] SevenSegmentDisplay
- [ ] SixteenSegmentDisplay 
- [ ] SquareRGBLed

image

### PR DESCRIPTION
**Pull request by SarahBornais**
Fixes #4 
Added documentation for each of the output circuit components.

### CODE REVIEW

**Comment by satu0king** [comment]
You are to embed actual circuits not images.

**Comment by SarahBornais** [comment]
The reason for using images rather than circuits was because I wanted to overlay annotations and there does not seem to be a "label" component in CircuitVerse (which I would recommend adding for teaching purposes in the future).

Additionally, circuit elements embedded in other docs on GitHub do not actually show up (you have to click a link to see them), and thus I did not think they would enrich the documentation that I created.

Here's a screenshot of the current docs that I see on a fully updated Chrome browser:
Screen Shot 2019-03-17 at 9 30 27 PM

If you would like, I will look in to embedding circuits in addition to the annotated images that I have included, and debugging why the embedded circuits in other docs do not show up properly.

**Comment by satu0king** [comment]
> The reason for using images rather than circuits was because I wanted to overlay annotations and there does not seem to be a "label" component in CircuitVerse (which I would recommend adding for teaching purposes in the future).
> 
> Additionally, circuit elements embedded in other docs on GitHub do not actually show up (you have to click a link to see them), and thus I did not think they would enrich the documentation that I created.
> 
> Here's a screenshot of the current docs that I see on a fully updated Chrome browser:
> Screen Shot 2019-03-17 at 9 30 27 PM
> 
> If you would like, I will look in to embedding circuits in addition to the annotated images that I have included, and debugging why the embedded circuits in other docs do not show up properly.

The iframes will not be rendered in your github page because it is not rendered by GitHub. But we do not care about GitHub, we care about docs.circuitverse.org where it will be rendered correctly. 

Also, we do have a text annotation element. 
image

File: docs/outputs.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+* [Square RGB LED](#square-rgb-led)
+* [Hex Display](#hex-display)
+* [Seven Segment Display](#seven-segment-display)
+* [Sixteen Segment Display](#sixteen-segment-display)
+
+This guide will explain how and when to use each of these components and provide some tips and tricks for doing so.
+
+## Output
+Output is the most simple output component. When connected to an input, it simply displays this input applied to it as a 0 or a 1.
+
```
Comment (satu0king) [inline docs/outputs.md]: You will have to illustrate with multiple bitwidths.

File: docs/outputs.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+## RGB LED
+Unlike the digital or variable LEDs, the RGB (red-green-blue) LED component can display a wide variety of colors. It takes three 8 bit inputs which describe the intensity of red, green, and blue color respectively. 
+
+![RGB Led Example](images/RGBled_example.png)
+
+If you want to display a certain color, the easiest way to find what the inputs need to be is to use a [color picker](https://htmlcolorcodes.com/color-picker/) to find the corresponding R, G, and B values and then use a [decimal-to-binary converter](https://codebeautify.org/decimal-binary-converter) (or do it manually for the math practice) to find the 8 bit value that you need to enter for each input.
+
+## Square RGB LED
+The square RGB LED is quite similar to the regular RGB LED, save the difference in shape. The main advantage to using this type of LED is that it allows you to simulate pixels more easily.
+
```
Comment (satu0king) [inline docs/outputs.md]: You can increase the length of the wires in SquareRGB, check the property menu for this.

**Comment by satu0king** [comment]
Add an entry in the `_sidebar.md` It should look something like this. 

```
* Circuit Elements
  * [CircuitVerse Outputs](outputs.md) 
  * [Standard Gates](gates.md)
  * [Decoders and Plexers](decodersandplexers.md)
  * [Splitter](splitter.md)
```
I saw your annotated images, they are quite good. We can probably have them along with the live circuit? You can reproduce the same effect by using the annotation element in CircuitVerse. 

This illustration of the 16 segment display is really good. 

I also like the gif of the variable LED. I am curious, how did you make this? 
image

**Comment by SarahBornais** [comment]
I appreciate your feedback. I will make the changes you requested by tomorrow. Thanks for showing me the text component, for some reason I just couldn't seem to find it. Since you're curious, I made the LED gif by taking screenshots at 16 bit increments and compiling them using an online gif creator.

**Comment by SarahBornais** [comment]
I made all of the changes that you requested. Let me know if there's anything that I missed or changed in a way that you don't like.

Additionally, I replaced the gif of the variable LED with an embedded circuit that uses a clock and counter to accomplish the same task. However, the tradeoff there is that the clock needs to go so fast for the LED intensity to change at a decent pace that flashes very quickly (people may find this annoying or distracting). Please take a look at the new variable LED circuit and let me know what you think. I still have the gif if you would like to change it back.

**Comment by satu0king** [comment]
Thanks, it looks good. Any reason for using constants instead of regular inputs? We are using regular inputs everywhere so I think it will be better to keep it uniform. 

Also, add the following to the top of the file. 
Contributing Authors: [@SarahBornais](https://github.com/SarahBornais/)

**Comment by SarahBornais** [comment]
You're completely right about the constants vs the inputs, I'll change that. I was worried that using an adder might look too overwhelming to demonstrate such a basic thing. However, now that I think about it, I will just use a subcircuit to abstract the complexity for the demonstration. I will make these changes by tomorrow.

**Comment by SarahBornais** [comment]
I made the changes you requested. Let me know if you find anything else that needs fixing.

As for the flashing clock/counter issue, I just ended up putting the clock/counter system in a subcircuit so that it doesn't irritate the user with its flashing, since I was going to have to create a subcircuit for the adder anyways if I went that route. I got it incrementing at what I think is a good pace.

**Comment by satu0king** [comment]
Very nice work, thanks.

**Pull request merged**